### PR TITLE
[Pytorch][Executorch] Handwritten view copy out ops should resize out

### PR DESCRIPTION
**Pull request by JacobSzwejbka**
Summary: Handwritten out ops should have feature parity with the codegend ones. This means they should resize out to the appropriate size. Q. Why are these handwritten instead of codegend anyway? Q2. Wheres a good spot to put the resize and copy helpers since they are reused in the codegend out kernels

Test Plan: ci.

Differential Revision: D42177051

### CODE REVIEW

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -350,6 +350,38 @@ namespace native {
 DEFINE_DISPATCH(cat_serial_stub);
 DEFINE_DISPATCH(stack_serial_stub);
 
+
+namespace {
```
Comment (larryliu0820) [inline aten/src/ATen/native/TensorShape.cpp:354]: Should we make this reusable by `CompositeViewCopyKernels.cpp`?

File: aten/src/ATen/native/TensorShape.cpp [hunk 2]
```diff
@@ -3846,56 +3877,64 @@ at::Tensor lift_fresh(const at::Tensor& self) {
 
 at::Tensor& _fw_primal_copy_out(const at::Tensor & self, int64_t level, at::Tensor & out) {
   auto tmp = self._fw_primal(level);
-  out.copy_(tmp);
+  resize_out_helper(out, tmp);
```
Comment (larryliu0820) [inline aten/src/ATen/native/TensorShape.cpp:3880]: Actually, I was thinking maybe we should autogen these ops. For example, if you add `autogen` to `_fw_primal_copy` entry [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L13259), I believe the we can delete this kernel.

File: aten/src/ATen/native/TensorShape.cpp (hunk 1 above)
Comment (JacobSzwejbka) [reply aten/src/ATen/native/TensorShape.cpp:354]: yes, wasnt sure where to put the util functions

Comment (larryliu0820) [reply aten/src/ATen/native/TensorShape.cpp:354]: So I guess it's cleaner to just autogen all these kernels if possible. It seems like most of them are having the same logic as the generated ones.

File: aten/src/ATen/native/TensorShape.cpp (hunk 2 above)
Comment (JacobSzwejbka) [reply aten/src/ATen/native/TensorShape.cpp:3880]: yeah I wasnt sure if there was some reason they were hand written, maybe they were added before the codegen thing existed. Ill just change the pr to delete and codegen these instead

**Comment by JacobSzwejbka** [comment]
Ok moved most of them to autogen. These two prs confused me on the remaining ones. https://github.com/pytorch/pytorch/pull/89000 and https://github.com/pytorch/pytorch/pull/84579. Basically not sure if this change would break @ezyang's _symint generation, and narrow_copy apparently has some legacy issues I think this might also break?

**Comment by larryliu0820** [comment]
Can you dedup the utils with [`CompositeViewCopyKernels.cpp`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/CompositeViewCopyKernels.cpp#L21-L61)

**Comment by larryliu0820** [comment]
Also curious if we have test coverage for these ops. Check `test_ops.py` to see if they are covered

**Comment by JacobSzwejbka** [comment]
> Can you dedup the utils with [`CompositeViewCopyKernels.cpp`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/CompositeViewCopyKernels.cpp#L21-L61)

If the symint ops and narrow_copy turn out fine to remove I may not need the utils at all and every op can be moved to codegen.

File: aten/src/ATen/native/TensorShape.cpp (hunk 1 above)
Comment (ezyang) [reply aten/src/ATen/native/TensorShape.cpp:354]: Just put them in a header. If you're not sure what header to put them in, make a fresh one! ViewCopy.h maybe?

**Review by ezyang — APPROVED** [review]
Love the deletion here

**Pull request closed without merging**